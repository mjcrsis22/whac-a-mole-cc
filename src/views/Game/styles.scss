.game-container {
  .game-section {
    margin-bottom: 0.5rem;
  }
}
