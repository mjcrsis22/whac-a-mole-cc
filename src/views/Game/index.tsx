import GameContextProvider from '../../contexts/GameContext';
import UserPanel from '../../components/UserPanel';
import GameBoard from '../../components/GameBoard';
import GameControls from '../../components/GameControls';
import GameSettingsPanel from '../../components/GameSettingsPanel';
import './styles.scss';

function Game() {
  return (
    <GameContextProvider>
      <div className="game-container">
        <div className="game-section">
          <UserPanel />
        </div>
        <div className="game-section">
          <GameBoard />
          <GameControls />
        </div>
        <div className="game-section">
          <GameSettingsPanel />
        </div>
      </div>
    </GameContextProvider>
  );
}

export default Game;
