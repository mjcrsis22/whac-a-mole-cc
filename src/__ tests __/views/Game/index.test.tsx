import { render } from '@testing-library/react';
import Game from '../../../views/Game';

jest.mock('../../../components/UserPanel', () => () => <>Cmp UserPanel</>);

describe('Game component', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });
  test('Renders correctly', () => {
    const { container } = render(<Game />);
    expect(container).toMatchSnapshot();
  });
});
