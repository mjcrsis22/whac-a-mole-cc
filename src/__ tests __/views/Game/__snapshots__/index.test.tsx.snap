// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game component Renders correctly 1`] = `
<div>
  <div
    class="game-container"
  >
    <div
      class="game-section"
    >
      Cmp UserPanel
    </div>
    <div
      class="game-section"
    >
      <div
        class="game-board-container"
      >
        <div
          class="play-container"
        >
          <div
            class="spot-wrapper"
            style="left: 5%; top: 0%;"
          >
            <div
              class="game-spot"
            >
              <div
                class="spot-mark "
              />
              <div
                class="spot-mole-wrapper"
              >
                <img
                  alt="Mole"
                  class="spot-mole "
                  role="button"
                  src="test-file-stub"
                />
              </div>
            </div>
          </div>
          <div
            class="spot-wrapper"
            style="left: 5%; top: 72%;"
          >
            <div
              class="game-spot"
            >
              <div
                class="spot-mark "
              />
              <div
                class="spot-mole-wrapper"
              >
                <img
                  alt="Mole"
                  class="spot-mole "
                  role="button"
                  src="test-file-stub"
                />
              </div>
            </div>
          </div>
          <div
            class="spot-wrapper"
            style="left: 30%; top: 15%;"
          >
            <div
              class="game-spot"
            >
              <div
                class="spot-mark "
              />
              <div
                class="spot-mole-wrapper"
              >
                <img
                  alt="Mole"
                  class="spot-mole "
                  role="button"
                  src="test-file-stub"
                />
              </div>
            </div>
          </div>
          <div
            class="spot-wrapper"
            style="left: 30%; top: 60%;"
          >
            <div
              class="game-spot"
            >
              <div
                class="spot-mark "
              />
              <div
                class="spot-mole-wrapper"
              >
                <img
                  alt="Mole"
                  class="spot-mole "
                  role="button"
                  src="test-file-stub"
                />
              </div>
            </div>
          </div>
          <div
            class="spot-wrapper"
            style="left: 45%; top: 40%;"
          >
            <div
              class="game-spot"
            >
              <div
                class="spot-mark "
              />
              <div
                class="spot-mole-wrapper"
              >
                <img
                  alt="Mole"
                  class="spot-mole "
                  role="button"
                  src="test-file-stub"
                />
              </div>
            </div>
          </div>
          <div
            class="spot-wrapper"
            style="left: 60%; top: 15%;"
          >
            <div
              class="game-spot"
            >
              <div
                class="spot-mark "
              />
              <div
                class="spot-mole-wrapper"
              >
                <img
                  alt="Mole"
                  class="spot-mole "
                  role="button"
                  src="test-file-stub"
                />
              </div>
            </div>
          </div>
          <div
            class="spot-wrapper"
            style="left: 60%; top: 60%;"
          >
            <div
              class="game-spot"
            >
              <div
                class="spot-mark "
              />
              <div
                class="spot-mole-wrapper"
              >
                <img
                  alt="Mole"
                  class="spot-mole "
                  role="button"
                  src="test-file-stub"
                />
              </div>
            </div>
          </div>
          <div
            class="spot-wrapper"
            style="left: 85%; top: 0%;"
          >
            <div
              class="game-spot"
            >
              <div
                class="spot-mark "
              />
              <div
                class="spot-mole-wrapper"
              >
                <img
                  alt="Mole"
                  class="spot-mole "
                  role="button"
                  src="test-file-stub"
                />
              </div>
            </div>
          </div>
          <div
            class="spot-wrapper"
            style="left: 85%; top: 72%;"
          >
            <div
              class="game-spot"
            >
              <div
                class="spot-mark "
              />
              <div
                class="spot-mole-wrapper"
              >
                <img
                  alt="Mole"
                  class="spot-mole "
                  role="button"
                  src="test-file-stub"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="game-controls-container"
      >
        <button
          class="button"
        >
          Play
        </button>
      </div>
    </div>
    <div
      class="game-section"
    >
      <div
        class="game-settings-panel"
      >
        <span>
          Levels
        </span>
        <div
          class="level-buttons"
        >
          <button
            class="button level-button selected"
            title="Level LOW"
          >
            LOW
          </button>
          <button
            class="button level-button "
            title="Level MEDIUM"
          >
            MEDIUM
          </button>
          <button
            class="button level-button "
            title="Level HARD"
          >
            HARD
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
